nixos/modules/services/monitoring/librenms.nix

```diff
@@ -0,0 +1,673 @@
+      wantedBy = [ "timers.target" ];
+      timerConfig = {
+        OnCalendar = "minutely";
+        AccuracySec = "1second";
+      };
+    };
+
+    systemd.services.librenms-setup = {
+      description = "Preparation tasks for LibreNMS";
+      before = [ "phpfpm-librenms.service" ];
+      after = [ "systemd-tmpfiles-setup.service" ]
+        ++ (optional (cfg.database.host == "localhost") "mysql.service");
+      wantedBy = [ "multi-user.target" ];
+      restartTriggers = [ package configFile ];
+      path = [ pkgs.mariadb pkgs.unixtools.whereis pkgs.gnused ];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        User = cfg.user;
+        Group = cfg.group;
+      };
+      script = ''
+        cat ${configFile} > ${cfg.dataDir}/config.php
+        ${optionalString (cfg.config.authMechanism == "ldap" && cfg.config.ldap.useBindUser) ''
+          echo -n "\$config['auth_ldap_bindpassword'] = '" >> ${cfg.dataDir}/config.php
+          cat ${cfg.config.ldap.bindPasswordFile} | tr -d "\n" >> ${cfg.dataDir}/config.php
+          echo "';" >> ${cfg.dataDir}/config.php
+        ''}
+        if [[ ! -s ${cfg.dataDir}/.env ]]; then
+          # init .env file
+          echo "APP_KEY=" > ${cfg.dataDir}/.env
+          ${artisanWrapper}/bin/artisan key:generate --ansi
+          ${artisanWrapper}/bin/artisan webpush:vapid
+          echo "" >> ${cfg.dataDir}/.env
+          echo -n "NODE_ID=" >> ${cfg.dataDir}/.env
+          ${phpPackage}/bin/php -r "echo uniqid();" >> ${cfg.dataDir}/.env
+          echo "" >> ${cfg.dataDir}/.env
+        else
+          # .env file already exists --> only update database config
+          ${pkgs.gnused}/bin/sed -i /^DB_/d ${cfg.dataDir}/.env
+        fi
+        echo "DB_HOST=${cfg.database.host}" >> ${cfg.dataDir}/.env
+        echo "DB_PORT=${toString cfg.database.port}" >> ${cfg.dataDir}/.env
+        echo "DB_DATABASE=${cfg.database.database}" >> ${cfg.dataDir}/.env
+        echo "DB_USERNAME=${cfg.database.username}" >> ${cfg.dataDir}/.env
+        echo -n "DB_PASSWORD=" >> ${cfg.dataDir}/.env
+        cat ${cfg.database.passwordFile} >> ${cfg.dataDir}/.env
+
+        # clear cache after update
+        OLD_VERSION=$(cat ${cfg.dataDir}/version)
+        if [[ $OLD_VERSION != "${package.version}" ]]; then
+          rm -r ${cfg.dataDir}/cache/*
+          echo "${package.version}" > ${cfg.dataDir}/version
+        fi
+
+        # convert rrd files when the oneMinutePolling option is changed
+        OLD_ENABLED=$(cat ${cfg.dataDir}/one_minute_enabled)
+        if [[ $OLD_ENABLED != "${boolToString cfg.enableOneMinutePolling}" ]]; then
+          ${phpPackage}/bin/php ${package}/scripts/rrdstep.php -h all
+          echo "${boolToString cfg.enableOneMinutePolling}" > ${cfg.dataDir}/one_minute_enabled
+        fi
+
+        # migrate db
+        ${phpPackage}/bin/php ${package}/includes/sql-schema/update.php
+      '';
+    };
+
+    programs.mtr.enable = true;
```

Comment:
Yes, because `programs.mtr` just creates a wrapper for `mtr-packet` which is then used by `mtr` itself --> `mtr` itself has no wrapper. But I have now changed the reference to `config.programs.mtr.package` to lower the confusion.

https://github.com/NixOS/nixpkgs/blob/nixos-23.05/nixos/modules/programs/mtr.nix#L34